spec/praxis/extensions/field_selection/active_record_query_selector_spec.rb

```diff
@@ -8,101 +8,142 @@
-      # Also, make AR do the actual query to make sure everything is wired up correctly
-      result = final_query.to_a
-      expect(result.size).to be > 2 # We are using 2 but we've seeded more
-      book1 = result[0]
-      book2 = result[1]
-      expect(book1.author.id).to eq 11
-      expect(book1.author.books.size).to eq 1
-      expect(book1.author.books.map(&:simple_name)).to eq(['Book1'])
-      expect(book1.category.name).to eq 'cat1'
-      expect(book1.tags.map(&:name)).to match_array(%w[blue red green])
+      it 'calls the select columns for the top level, and includes the right association hashes' do
+        expected_includes = {
+          author: {
+            books: {
+            },
+          },
+          category: {
+            books: {
+            },
+          },
+          tags: {
+          },
+        }
+        # expect(query).to receive(:includes).with(expected_includes).and_return(query)
+        expect(subject).to_not receive(:explain_query)
+        final_query = subject.generate
+        expect(final_query.select_values).to match_array(
+          expected_select_from_to_query,
+        )
+        # Our query selector always uses a single hash tree from the top, not an array of things
+        includes_hash = final_query.includes_values.first
+        expect(includes_hash).to match(expected_includes)
+        # Also, make AR do the actual query to make sure everything is wired up correctly
+        result = final_query.to_a
+        expect(result.size).to be > 2 # We are using 2 but we've seeded more
+        book1 = result[0]
+        book2 = result[1]
+        expect(book1.author.id).to eq 11
+        expect(book1.author.books.size).to eq 1
+        expect(book1.author.books.map(&:simple_name)).to eq(['Book1'])
+        expect(book1.category.name).to eq 'cat1'
+        expect(book1.tags.map(&:name)).to match_array(%w[blue red green])
 
-      expect(book2.author.id).to eq 22
-      expect(book2.author.books.size).to eq 1
-      expect(book2.author.books.map(&:simple_name)).to eq(['Book2'])
-      expect(book2.category.name).to eq 'cat2'
-      expect(book2.tags.map(&:name)).to match_array(['red'])
-    end
+        expect(book2.author.id).to eq 22
+        expect(book2.author.books.size).to eq 1
+        expect(book2.author.books.map(&:simple_name)).to eq(['Book2'])
+        expect(book2.category.name).to eq 'cat2'
+        expect(book2.tags.map(&:name)).to match_array(['red'])
+      end
 
-    it 'calls the explain debug method if enabled' do
-      suppress_output do
-        # Actually make it run all the way...but suppressing the output
-        subject.generate
+      it 'calls the explain debug method if enabled' do
+        suppress_output do
+          # Actually make it run all the way...but suppressing the output
+          subject.generate
+        end
+      end
+
+      context 'and fields that contain transitive dependencies on the root model' do
```

Comment:
I think, spec-wise, it'd be clearer if we make 2 'tests' within this context:
1- that shows that only  the top level requested fields are added
2- one that shows that when adding the transitive dept do it, new fields show up.

```
For 1:
let(:selector_fields) { { simple_name: true, author: true } }
> top level has simple_name (and author_id and added column)
For 2:
let(:selector_fields) { { simple_name: true, author: { books: { isbn: true } } } }
> now Top level has the same as 1, PLUS isbn
```